openml/extensions/sklearn/extension.py

```diff
@@ -860,7 +868,16 @@ def flatten_all(list_):
                         # length 3 is for ColumnTransformer
                         msg = 'Length of tuple does not match assumptions'
                         raise ValueError(msg)
-                    if not isinstance(sub_component, (OpenMLFlow, type(None))):
+
+                    if isinstance(sub_component, str):
```

Comment:
This makes sense, but could you please add an explanation of which scikit-learn model these support?

Reply:
I shall add the explanations based on the example I know. However, as I mentioned in the PR comment, I am not sure if these changes/assumptions can generalize to ALL such sklearn models.
I don't think we need to handle cases wherein sklearn cannot process a user input, that is, a str in place of an estimator. For a `ColumnTransform` sklearn handles the keywords _drop_ and _passthrough_ by treating them as estimators that are not evaluated. Hence, I am explicitly handling this case.
Should I forage the sklearn docs a bit more to see if any such models exist?

Reply:
I don't know any other examples myself, so testing the example from the issue plus maybe one or two examples from the scikit-learn docs should be sufficient. I don't think you need to search for other scikit-learn models which allow strings, though. Maybe @amueller has some additional input?

Reply:
Sure, will do so.